crates/core/component/auction/src/component/action_handler/dutch/end.rs

```diff
@@ -12,7 +16,26 @@ impl ActionHandler for ActionDutchAuctionEnd {
         Ok(())
     }
 
-    async fn check_and_execute<S: StateWrite>(&self, mut _state: S) -> Result<()> {
+    async fn check_and_execute<S: StateWrite>(&self, mut state: S) -> Result<()> {
+        let auction_id = self.auction_id;
+
+        let auction_state = state
+            .get_dutch_auction_by_id(auction_id)
+            .await
+            .context("the auction associated with this id is not a dutch auction")?;
+
+        let Some(auction) = auction_state else {
+            bail!("no auction found for id {auction_id}")
+        };
+
+        // Check that the sequence number for the auction state is 0 (opened) or 1 (closed).
+        ensure!(
+            auction.state.sequence <= 1,
```

Comment:
tioli: i might write this to directly check that `sequence == 0 || sequence == 1`, or `matches!(sequence, 0 | 1)`

Reply:
sg